route/build/src/datetime_data.f90

```diff
@@ -443,4 +465,35 @@ logical(lgt) FUNCTION fn_is_le(this, that)
     end if
   END FUNCTION fn_is_le
 
+  real(dp) FUNCTION fn_delta(this, that) result(ans)
+    ! this - that give in time difference in second
+    implicit none
+    class(datetime),     intent(in)  :: this
+    class(datetime),     intent(in)  :: that
+    ! local variables
+    real(dp)                         :: sec_in_day1
+    real(dp)                         :: sec_in_day2
+    real(dp)                         :: diffDay
+    integer(i4b)                     :: ierr
+    type(datetime)                   :: date1
+    type(datetime)                   :: date2
+    real(dp)                         :: julday1
+    real(dp)                         :: julday2
+    character(len=strLen)            :: cmessage
+
+    sec_in_day1 = real(this%ih*3600+this%imin*60, kind=dp) +this%dsec
+    sec_in_day2 = real(that%ih*3600+that%imin*60, kind=dp) +that%dsec
+
+    date1 = datetime(this%iy, this%im, this%id, 0, 0, 0.0, calendar=this%calendar)
+    date2 = datetime(that%iy, that%im, that%id, 0, 0, 0.0, calendar=that%calendar)
+
+    call date1%julianday(julday1, ierr, cmessage)
+    call date2%julianday(julday2, ierr, cmessage)
+
+    diffDay=julday1-julday2
+
+    ans = diffDay*86400._dp+sec_in_day1-sec_in_day2
```

Comment:
86400 would be better as a parameter

Reply:
yes, there is already parameter for this (`secprday`), so replace this with the parameter